test/generator/exercise_test.rb

```diff
@@ -2,35 +2,43 @@
 
 module Generator
   class ExerciseTest < Minitest::Test
-    def test_version
-      mock_version_file = Minitest::Mock.new.expect :to_i, 1
-      mock_repository = Minitest::Mock.new.expect :tests_version, mock_version_file
+    FixturePaths = Paths.new(
+      metadata: 'test/fixtures/metadata',
+      track: 'test/fixtures/xruby'
+    )
 
-      subject = Exercise.new(repository: mock_repository)
+    def test_version
+      repository = Repository.new(paths: FixturePaths, slug: 'alpha')
+      subject = Exercise.new(repository: repository)
       assert_equal 1, subject.version
-
-      mock_repository.verify
-      mock_version_file.verify
     end
 
     def test_update_tests_version
-      mock_version_file = Minitest::Mock.new
-      mock_version_file.expect :increment, 2
-      mock_repository = Minitest::Mock.new.expect :tests_version, mock_version_file
-
-      subject = Exercise.new(repository: mock_repository)
-      assert_equal 2, subject.update_tests_version
-
-      mock_repository.verify
-      mock_version_file.verify
+      repository = Repository.new(paths: FixturePaths, slug: 'alpha')
+      # Verify iniital condition from fixture file
+      assert_equal 1, repository.tests_version.to_i
+
+      mock_file = Minitest::Mock.new.expect :write, '2'.length, [2]
+      subject = Exercise.new(repository: repository)
+      File.stub(:open, true, mock_file) do
+        assert_equal 2, subject.update_tests_version
+      end
+      mock_file.verify
```

Comment:
I actually had that version of these tests, but ran into issues with the mocking. I may try a second version...